include/rmm/mr/device/arena_memory_resource.hpp

```diff
@@ -97,7 +97,10 @@ class arena_memory_resource final : public device_memory_resource {
     : global_arena_{upstream_mr, arena_size}, dump_log_on_failure_{dump_log_on_failure}
   {
     if (dump_log_on_failure_) {
-      logger_ = spdlog::basic_logger_mt("arena_memory_dump", "rmm_arena_memory_dump.log");
+      logger_ =
+        std::make_shared<spdlog::logger>("arena_memory_dump",
+                                         std::make_shared<spdlog::sinks::basic_file_sink_mt>(
```

Comment:
*question*: Why this change?

Reply:
It's changed to avoid exposing a spdlog symbol to the ABI. I assume the basic_logger_mt uses a global map to store loggers with default visibility.
The truncate file is the same as the logging resource, just following the convention there.

Reply:
Does `spdlog::basic_logger_mt` not just return a `shared_ptr<spdlog::logger>`? Which symbol are we now not exposing in the ABI?

Reply:
Please see this test failure: https://github.com/rapidsai/rmm/actions/runs/11532359492/job/32104141080

Reply:
Ah, thanks.